### ISSUE TITLE
Enable option to not initialise reply-to queue.

### ISSUE THREAD

**Issue by fechy** [golevelup/nestjs#109]
Since it is by default that the reply-to queue is being listened to, it is impossible to use only subscribe if the RabbitMQ node doesn't have 'amq.rabbitmq.reply-to'.

The node I'm trying to connect to is not using this reply-to, therefore this error is thrown on start:
```bash
Error: Channel closed by server: 404 (NOT-FOUND) with message "NOT_FOUND - no queue 'amq.rabbitmq.reply-to' in vhost '/'"
    at Channel.C.accept (/Users/fechy/projects/coliquio/tagging-api/node_modules/amqplib/lib/channel.js:422:17)
    at Connection.mainAccept [as accept] (/Users/fechy/projects/coliquio/tagging-api/node_modules/amqplib/lib/connection.js:64:33)
    at Socket.go (/Users/fechy/projects/coliquio/tagging-api/node_modules/amqplib/lib/connection.js:478:48)
    at emitNone (events.js:106:13)
    at Socket.emit (events.js:208:7)
    at emitReadable_ (_stream_readable.js:513:10)
    at emitReadable (_stream_readable.js:507:7)
    at addChunk (_stream_readable.js:274:7)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at TCP.onread (net.js:597:20)
```

One alternative will be to add an option to disable this initialization, as an "optOut".

**Comment by zerobig** [comment]
What version of RabbitMQ server?
Support the "direct reply-to" extension was introduced in RabbitMQ 3.4 (21 Oct 2014)
5 years... All support periods are ended.

**Comment by fechy** [comment]
The server is 3.6, it should support it, but I believe it was disabled by the admins. Problem is that I don't have control over this server and can only consume messages. 
Unfortunately, this means if I cannot disable this option from the library, I cannot use the library.

**Comment by WonderPanda** [comment]
@fechy It should be pretty straightforward to conditionally initialize the Rely To Queue based on the module configuration. This line would just need to be wrapped in a condition based on the configuration object:
https://github.com/golevelup/nestjs/blob/c20ba1b67eb3855f4c2b10d3e625891eb7a14042/packages/rabbitmq/src/amqp/connection.ts#L155

Perhaps you'd like to take a crack at making a contribution to the library? If not I think this is a reasonable thing to add but I'm busy in the evenings with a new product that I'm trying to launch so it might be a few days before I get around to implementing this

**Comment by fechy** [comment]
Sounds fair. I'll get to it.

**Comment by WonderPanda** [comment]
That's awesome! Thanks fechy, feel free to throw questions in here if you come up against anything you're not sure of

**Comment by WonderPanda** [comment]
@fechy Did you ever get a chance to take a crack at implementing this feature?

**Comment by fechy** [comment]
@WonderPanda I'm afraid not. I'll take a look as soon as I got some time back for my own.

**Comment by WonderPanda** [comment]
@fechy No worries. I may have some time to implement this today

**Comment by WonderPanda** [comment]
@fechy PR is here if you want to take a look: https://github.com/golevelup/nestjs/pull/114

**Comment by WonderPanda** [comment]
@fechy This is now available in `@golevelup/nestjs-rabbitmq@1.12.0`
